git/gitlab.go

```diff
@@ -246,7 +246,7 @@ func (g *GitLab) GetArchiveLink(project, tag string) (*url.URL, error) {
 	}
 
 	u, err := url.Parse(
-		fmt.Sprintf("/%s/repository/archive.tar.gz?ref=%s&private_token=%s",
+		fmt.Sprintf("/api/v4/projects/%s/repository/archive.tar.gz?ref=%s&private_token=%s",
```

Comment:
Fair point, I'll make the changes

Reply:
When you do, you should undo the `url.QueryEscape(ns)` change as that is not needed in this URL.

Reply:
I read the above too quickly. The issue here is that all GET requests that do not go to `/api/v4` without a valid cookie (even if you have an access token) are now returning 401 Unauthorized errors. Changing the URL to use the API is now the only way to download the archive file.

Reply:
I see... In that case I suggest to make the URL relative instead of absolute and let the URL package take care of resolving the correct URL (using the call at the end of this package `g.client.BaseURL().ResolveReference(u)`):

```go
fmt.Sprintf("projects/%s/repository/archive.tar.gz?sha=%s&private_token=%s",
  url.QueryEscape(ns),
  tag,
  g.token,
)
```

Als note that `ref` needs to be changed to `sha` here ^^